mlir/lib/Conversion/VectorToArmSME/VectorToArmSME.cpp

```diff
@@ -356,6 +356,20 @@ struct TransposeOpToArmSMELowering
       return failure();
 
     auto loc = transposeOp.getLoc();
+    Value input = transposeOp.getVector();
+
+    if (auto xferOp = input.getDefiningOp<vector::TransferReadOp>()) {
+      // Fold transpose into transfer_read to enable in-flight transpose when
+      // converting to arm_sme.tile_load.
```

Comment:
This needs to check the `xferOp` only has one user (or all users are the same transpose), otherwise this may not be valid.